discord/client.py

```diff
@@ -399,9 +399,8 @@ def _schedule_event(
         # Schedules the task
         return asyncio.create_task(wrapped, name=f"pycord: {event_name}")
 
-    def dispatch(self, event: str, *args: Any, **kwargs: Any) -> None:
+    def dispatch(self, event: Union[str, EventType], *args: Any, **kwargs: Any) -> None:
         _log.debug("Dispatching event %s", event)
```

Comment:
Should `event` be cast to `str` here?

Reply:
Can you elaborate?

Reply:
> Can you elaborate?

If an `EventType` is passed it'll stay as one for the entire method, as well as in this:
`coro = self.events.get(event)`
And events are being stored as strings

Reply:
I really don't understand why this is a problem. I still want to keep support for custom events, so I do this by allowing a `str` and a `EventType` types to be passed in.